text/0087-graphql-errors.md

```diff
@@ -0,0 +1,262 @@
+* Start Date: 2023-05-17
+* RFC Type: feature
+* RFC PR: [#93](https://github.com/getsentry/rfcs/pull/93)
+* RFC Status: draft
+* RFC Driver: [Manoel Aranda Neto](https://github.com/marandaneto)
+
+# Summary
+
+Add `Request` and `Response` body to events.
+
+This feature is opt-in by default due to PII concerns.
+
+# Motivation
+
+The [Request](https://develop.sentry.dev/sdk/event-payloads/types/#request) interface contains information on a HTTP request related to the event.
+
+The `Request` contains the `data` field which is the request body. Can be given as string or structural data of any format.
+
+The [Response](https://develop.sentry.dev/sdk/event-payloads/types/#responsecontext) interface contains information on a HTTP response related to the event.
+
+However, the `Response` interface does not contain the `data` field which is the response body.
+
+# Background
+
+This is necessary to implement the [GraphQL Client Errors](https://www.notion.so/sentry/GraphQL-proposal-d6e5846f30434770903cf3af20bc2568) with syntax highlight.
+
+The `Request` and `Response` bodies could contain PII.
+
+Request body example (query language only):
+
+```graphql
+query { 
+  viewer { 
+    login
+  }
+}
+```
+
+Response body example:
+
+```json
+{
+  "data": {
+    "viewer": {
+      "login": "foo"
+    }
+  }
+}
+```
+
+A Request example with an error (query language only):
+
+```graphql
+query { 
+  viewer { 
+    # note the removed `n` at the end
+    logi
+  }
+}
+```
+
+Response body example:
+
+```json
+{
+  "errors": [
+    {
+      "path": [
+        "query",
+        "viewer",
+        "logi"
+      ],
+      "extensions": {
+        "code": "undefinedField",
+        "typeName": "User",
+        "fieldName": "logi"
+      },
+      "locations": [
+        {
+          "line": 4,
+          "column": 5
+        }
+      ],
+      "message": "Field 'logi' doesn't exist on type 'User'"
+    }
+  ]
+}
+```
+
+Using the `locations` and the request body (query language only), we can highlight the error in the request body.
+
+```graphql
+query { 
+  viewer { 
+    # note the removed `n` at the end
+    -> logi
+  }
+}
+```
+
+Request body example, full body (not only the query language):
+
+```json
+{
+  "query": "{\n  viewer {\n    login\n  }\n}",
+  "variables": {}
+}
+```
+
+Request body example, full body (not only the query language):
+
+```json
+{
+  "query": "{\n  viewer {\n    login\n  }\n}",
+  "variables": {}
+}
+```
```

Comment:
`l`: This section is duplicated if I'm not mistaken.
```suggestion
```

Reply:
Good catch, I guess it was a merging or copypasta error.